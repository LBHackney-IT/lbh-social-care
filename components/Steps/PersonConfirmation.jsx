import PropTypes from 'prop-types';
import { useRouter } from 'next/router';
import Button from 'components/Button/Button';

const PersonConfirmation = ({ formData, formSteps }) => {
  const router = useRouter();
  const { personId } = router.query;
  if (!formSteps) return null;
  return (
    <>
      <h1 className="govuk-fieldset__legend--l gov-weight-lighter govuk-!-margin-bottom-7">
        Add new person confirmed
      </h1>
      <div className="govuk-panel govuk-panel--confirmation govuk-!-margin-bottom-9">
        <h1 className="govuk-fieldset__legend--l gov-weight-lighter">
          {formData.firstName} {formData.lastName} has been added.
          <br />#{personId}
        </h1>
      </div>
      <div>
        <Button label="View person" isSecondary route={`/people/${personId}`} />
      </div>
    </>
  );
};

PersonConfirmation.propTypes = {
  formData: PropTypes.shape({}).isRequired,
  formSteps: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
};

export default PersonConfirmation;
