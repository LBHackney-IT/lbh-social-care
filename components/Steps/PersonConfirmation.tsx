import { useRouter } from 'next/router';

import Button from 'components/Button/Button';

interface Props {
  formData: Record<string, unknown>;
  successMessage?: string;
}

const referralForm =
  'https://docs.google.com/forms/d/e/1FAIpQLSc4y4yjw6DpwWHN2fias1SEDo0lZZZNgN3M20Zy_p2He1rSuw/viewform';

const PersonConfirmation = ({
  formData,
  successMessage,
}: Props): React.ReactElement => {
  const { query, asPath } = useRouter();
  const { ref } = query;
  return (
    <>
      {successMessage && (
        <h1 className="govuk-fieldset__legend--l gov-weight-lighter govuk-!-margin-bottom-7">
          {successMessage}
        </h1>
      )}
      <div className="govuk-panel govuk-panel--confirmation govuk-!-margin-bottom-9">
        <h1 className="govuk-fieldset__legend--l gov-weight-lighter">
          {formData.firstName} {formData.lastName} has been{' '}
          {asPath?.includes('add') ? 'added' : 'updated'}.
          <br />#{ref}
        </h1>
      </div>
      <div style={{ display: 'flex', justifyContent: 'space-between' }}>
        <Button
          label="View person"
          isSecondary
          wideButton
          route={`/people/${ref}`}
        />
        <Button label="Add referral details" wideButton route={referralForm} />
      </div>
    </>
  );
};

export default PersonConfirmation;
