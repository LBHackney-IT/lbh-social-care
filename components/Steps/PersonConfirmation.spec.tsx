import { render } from '@testing-library/react';
import PersonConfirmation from './PersonConfirmation';

jest.mock('next/router', () => ({
  useRouter: () => ({
    query: { ref: '123456' },
  }),
}));

describe('Person Confirmation component', () => {
  const props = {
    formData: {
      bar_input: 'foo',
      firstName: 'first',
      lastName: 'last',
    },
    formSteps: [
      {
        components: [
          {
            component: 'TextInput',
            name: 'bar_input',
            width: 30,
            label: 'Foo',
          },
        ],
        id: 'foo-bar',
        title: 'Foo Bar',
      },
    ],
  };
  it('should render properly', () => {
    const { asFragment } = render(<PersonConfirmation {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
