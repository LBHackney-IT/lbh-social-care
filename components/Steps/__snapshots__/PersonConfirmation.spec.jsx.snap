// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Person Confirmation component should render properly 1`] = `
<DocumentFragment>
  <h1
    class="govuk-fieldset__legend--l gov-weight-lighter govuk-!-margin-bottom-7"
  >
    Add new person confirmed
  </h1>
  <div
    class="govuk-panel govuk-panel--confirmation govuk-!-margin-bottom-9"
  >
    <h1
      class="govuk-fieldset__legend--l gov-weight-lighter"
    >
      first last has been added.
      <br />
      #123456
    </h1>
  </div>
  <div
    style="display: flex; justify-content: space-between;"
  >
    <button
      class="govuk-button lbh-button govuk-button--secondary lbh-button--secondary"
      data-module="govuk-button"
      style="min-width: 200px;"
    >
      View person
    </button>
    <button
      class="govuk-button lbh-button"
      data-module="govuk-button"
      style="min-width: 200px;"
    >
      Add referral details
    </button>
  </div>
</DocumentFragment>
`;
